java-checks-test-sources/src/main/java/checks/regex/ImpossibleBackReferenceCheck.java

```diff
@@ -41,9 +41,12 @@ void compliant(String str) {
     str.matches("(?:\\1|x(.)){1,2}");
     str.matches("(?:\\1\\2|(x)(.))*");
     str.matches("(.)(?:\\1\\2|x(.))*");
+    str.matches("(1)\\11"); // Compliant, backreference is \1 because group 11 does not exist
+    str.matches("(1)(2)(3)(4)(5)(6)(7)(8)(9)(a)\\11(b)"); // Compliant, backreference is \1 because 11 does not exist at this point in the regex
+    str.matches("(1)(2)(3)(4)(5)(6)(7)(8)(9)(a)(b)\\11"); // Compliant, backreference is \11 because group 11 exists at this point in the regex
   }
 
-  @org.hibernate.validator.constraints.URL(regexp = "\\1(.)") // Noncompliant [[sc=54;ec=57;secondary=46]] {{Fix this backreference, so that it refers to a group that can be matched before it.}}
+  @org.hibernate.validator.constraints.URL(regexp = "\\1(.)") // Noncompliant [[sc=54;ec=57;secondary=49]] {{Fix this backreference, so that it refers to a group that can be matched before it.}}
```

Comment:
Since you heroically implemented relative secondary locations, we should replace absolute secondaries with `secondary=+0` whenever we have to change line numbers, so it's the last time we have to change that particular line number.

Reply:
I should have remembered it :man_facepalming: